backend/de.metas.business/src/main/java/de/metas/project/budget/interceptor/C_Project_Resource_Budget.java

```diff
@@ -40,23 +44,25 @@
 public class C_Project_Resource_Budget
 {
 	private final MultiCalendarService multiCalendarService;
+	private final BudgetProjectResourceService budgetProjectResourceService;
+	private final ResourceService resourceService;
 
 	public C_Project_Resource_Budget(
-			final MultiCalendarService multiCalendarService)
+			@NonNull final MultiCalendarService multiCalendarService,
+			@NonNull final BudgetProjectResourceService budgetProjectResourceService,
+			@NonNull final ResourceService resourceService)
 	{
 		this.multiCalendarService = multiCalendarService;
+		this.budgetProjectResourceService = budgetProjectResourceService;
+		this.resourceService = resourceService;
 	}
 
-	@ModelChange(timings = { ModelValidator.TYPE_BEFORE_NEW, ModelValidator.TYPE_BEFORE_CHANGE })
-	public void beforeSave(final I_C_Project_Resource_Budget record)
+	@ModelChange(timings = { ModelValidator.TYPE_AFTER_NEW, ModelValidator.TYPE_AFTER_CHANGE })
+	public void afterSave(final I_C_Project_Resource_Budget record, final ModelChangeType changeType)
```

Comment:
do you still need `, final ModelChangeType changeType` param?

Reply:
Yes, it's still needed in de.metas.project.budget.interceptor.C_Project_Resource_Budget#notifyIfUserChange